File: frontend/lib/types/task-collaboration.ts
import type { Task } from "@/lib/types";

export interface TaskComment {
  id: string;
  task_id: string;
  company_id: string;
  agent_id: string;
  content: string;
  created_at: string;
}

export interface TaskDependency {
  id: string;
  task_id: string;
  depends_on_id: string;
  company_id: string;
  created_at: string;
}

export interface TaskWatcher {
  task_id: string;
  agent_id: string;
  created_at: string;
}

export interface TaskDetail extends Task {
  tags: string[];
  comments: TaskComment[];
  dependencies: TaskDependency[];
  watchers: TaskWatcher[];
}

File: backend/internal/domain/task_collaboration.go
package domain

import "time"

type TaskComment struct {
	ID        string    `gorm:"column:id"         json:"id"`
	TaskID    string    `gorm:"column:task_id"    json:"taskId"`
	CompanyID string    `gorm:"column:company_id" json:"companyId"`
	AgentID   string    `gorm:"column:agent_id"   json:"agentId"`
	Content   string    `gorm:"column:content"    json:"content"`
	CreatedAt time.Time `gorm:"column:created_at" json:"createdAt"`
}

type TaskDependency struct {
	ID          string    `gorm:"column:id"            json:"id"`
	TaskID      string    `gorm:"column:task_id"       json:"taskId"`
	DependsOnID string    `gorm:"column:depends_on_id" json:"dependsOnId"`
	CompanyID   string    `gorm:"column:company_id"    json:"companyId"`
	CreatedAt   time.Time `gorm:"column:created_at"    json:"createdAt"`
}

type TaskWatcher struct {
	TaskID    string    `gorm:"column:task_id"    json:"taskId"`
	AgentID   string    `gorm:"column:agent_id"   json:"agentId"`
	CreatedAt time.Time `gorm:"column:created_at" json:"createdAt"`
}
